### PR TITLE
Refresh plugin for August 2023 (Upgrade to a minimum core version 2.387.3 and minimum Java version 11)

### PR DESCRIPTION
**Pull request by basil**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -18,17 +18,17 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>4.45</version>
+    <version>4.67</version>
```
Comment (basil) [inline pom.xml]: Latest version at the time of this writing.

File: pom.xml [hunk 2]
```diff
@@ -18,17 +18,17 @@
...
     <relativePath/>
   </parent>
 
   <artifactId>google-compute-engine</artifactId>
-  <version>4.3.16-SNAPSHOT</version>
+  <version>${revision}${changelist}</version>
```
Comment (basil) [inline pom.xml:26]: Following the archetype.

File: pom.xml [hunk 3]
```diff
@@ -18,17 +18,17 @@
...
   <packaging>hpi</packaging>
 
   <name>Google Compute Engine Plugin</name>
   <description>This plugin allows Jenkins to provision dynamic worker nodes on Google Compute Engine.</description>
-  <url>https://github.com/jenkinsci/google-compute-engine-plugin</url>
+  <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
```
Comment (basil) [inline pom.xml:31]: Following the archetype.

File: pom.xml [hunk 4]
```diff
@@ -62,13 +62,11 @@
   </scm>
 
   <properties>
-    <revision>4.3.14</revision>
+    <revision>4.3.16</revision>
     <changelist>-SNAPSHOT</changelist>
-    <gitHubRepo>jenkinsci/google-compute-engine-plugin</gitHubRepo>
-    <jenkins.version>2.346.1</jenkins.version>
+    <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
```
Comment (basil) [inline pom.xml:67]: Following the archetype.

File: pom.xml [hunk 5]
```diff
@@ -164,11 +156,6 @@
       <groupId>io.jenkins.plugins</groupId>
       <artifactId>commons-text-api</artifactId>
     </dependency>
-    <dependency>
-      <groupId>commons-io</groupId>
-      <artifactId>commons-io</artifactId>
-      <version>2.8.0</version>
-    </dependency>
```
Comment (basil) [inline pom.xml:171]: Part of Jenkins core, so does not need to be delivered in this plugin.

File: pom.xml [hunk 6]
```diff
@@ -219,7 +206,6 @@
     <dependency>
       <groupId>org.hamcrest</groupId>
       <artifactId>hamcrest</artifactId>
-      <version>${hamcrest.version}</version>
```
Comment (basil) [inline pom.xml:217]: Defined in the plugin parent POM.

**Comment by basil** [comment]
Currently blocking https://github.com/jenkinsci/bom/pull/2440

File: pom.xml [hunk 7]
```diff
@@ -62,13 +62,11 @@
...
-    <revision>4.3.14</revision>
+    <revision>4.3.17</revision>
     <changelist>-SNAPSHOT</changelist>
-    <gitHubRepo>jenkinsci/google-compute-engine-plugin</gitHubRepo>
-    <jenkins.version>2.346.1</jenkins.version>
+    <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
+    <jenkins.version>2.387.3</jenkins.version>
     <jenkinsRuleTimeout>0</jenkinsRuleTimeout>
-    <maven.compiler.source>1.8</maven.compiler.source>
-    <maven.compiler.target>1.8</maven.compiler.target>
```
Comment (olamy) [inline pom.xml:71]: maybe worth to add the minimum jdk requirement in the PR title as is this will be in the release notes?

**Review by olamy — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 7 above)
Comment (basil) [reply pom.xml:71]: Am I understanding correctly that you are a maintainer of this repository and can therefore edit PR titles?

**Comment by basil** [comment]
Currently blocking https://github.com/jenkinsci/bom/pull/2440

File: pom.xml (hunk 7 above)
Comment (olamy) [reply pom.xml:71]: > Am I understanding correctly that you are a maintainer of this repository and can therefore edit PR titles?

I''m really sorry for the delay and misunderstanding. But as you are the owner of the current work (this PR) I was under the impression you would have been the most appropriate to modify your own work after review from the member of this project's community. I'm not usually modifying work from others.
But so as it looks to take time, I will do it.

**Comment by olamy** [comment]
I take the opportunity to modify the title. I hope this doesn't alter too much the content of your work.

**Comment by basil** [comment]
Thanks. I generally don't include minimum core version changes in release notes for the plugins I maintain (unless the minimum core version is unusual, like a weekly release or a very recent LTS) because such plugin releases aren't even advertised in the UI to users of older cores. Similarly, a recent core version implies a minimum Java version, and users running older Java versions (and therefore older cores) wouldn't get an advertisement for such a plugin upgrade in the UI, so I generally don't include minimum Java version changes in release notes. I have filed dozens (hundreds?) of PRs without this information in the title and shipped dozens of releases without this information in the release notes, all without any complaints from users or other developers. But I acknowledge that as the plugin maintainer, you should be free to include (or not include) any information you want in the release notes. I hope that this discussion about writing release notes has not been a significant distraction from the comparatively far more laborious task of modernizing the plugin itself.

**Pull request merged**